solutions/agents/variables.tf

```diff
@@ -0,0 +1,222 @@
+  type        = list(string)
+  description = "List of tags to associate to all log records that the agent collects so that you can identify the agent's data quicker in the logging UI. NOTE: Use the 'log_analysis_add_cluster_name' variable to add the cluster name as a tag."
+  default     = []
+  nullable    = false
+}
+
+variable "log_analysis_add_cluster_name" {
+  type        = bool
+  description = "If true, configure the log analysis agent to attach a tag containing the cluster name to all log messages."
+  default     = true
+}
+
+variable "log_analysis_ingestion_key" {
+  type        = string
+  description = "Ingestion key for the IBM Cloud Logging agent to communicate with the instance."
+  sensitive   = true
+  default     = null
+}
+
+variable "log_analysis_secret_name" {
+  type        = string
+  description = "The name of the secret which will store the ingestion key."
+  default     = "logdna-agent"
+  nullable    = false
+}
+
+variable "log_analysis_instance_region" {
+  type        = string
+  description = "The IBM Log Analysis instance region. Used to construct the ingestion endpoint."
+  default     = "us-south"
+}
+
+variable "log_analysis_endpoint_type" {
+  type        = string
+  description = "Specify the IBM Log Analysis instance endpoint type (public or private) to use. Used to construct the ingestion endpoint."
+  default     = "private"
+  validation {
+    error_message = "The specified endpoint_type can be private or public only."
+    condition     = contains(["private", "public"], var.log_analysis_endpoint_type)
+  }
+}
+
+variable "log_analysis_agent_custom_line_inclusion" {
+  description = "Log Analysis agent custom configuration for line inclusion setting LOGDNA_K8S_METADATA_LINE_INCLUSION. See https://github.com/logdna/logdna-agent-v2/blob/master/docs/KUBERNETES.md#configuration-for-kubernetes-metadata-filtering for more info."
+  type        = string
+  default     = "label.app.kubernetes.io/name:sample-app\\, annotation.user:sample-user"
+}
+
+variable "log_analysis_agent_custom_line_exclusion" {
+  description = "Log Analysis agent custom configuration for line exclusion setting LOGDNA_K8S_METADATA_LINE_EXCLUSION. See https://github.com/logdna/logdna-agent-v2/blob/master/docs/KUBERNETES.md#configuration-for-kubernetes-metadata-filtering for more info."
+  type        = string
+  default     = "namespace:default"
+}
+
+variable "log_analysis_agent_name" {
+  description = "Log Analysis agent name. Used for naming all kubernetes and helm resources on the cluster."
+  type        = string
+  default     = "logdna-agent"
+  nullable    = false
+}
+
+variable "log_analysis_agent_namespace" {
+  type        = string
+  description = "Namespace where to deploy the Log Analysis agent. Default value is 'ibm-observe'."
+  default     = "ibm-observe"
+  nullable    = false
+}
+
+variable "log_analysis_agent_tolerations" {
+  description = "List of tolerations to apply to Log Analysis agent."
```

Comment:
Can you also add this: 
Defaults to the Exists operator meaning it will match any taint on any node

Reply:
Added